spec/injector_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'injector'
+
+describe Array do
+  describe "Array#injector" do
+
+    it "can take a block as an argument" do
+      expect([1].injector{"this is a block"}).to eq("this is a block")
+    end
+
+    it "can be called without a block" do
+      expect{[1].injector}.not_to raise_error
+    end
+
+    it "can be passed an initial value" do
+      expect([1].injector(2)).to eq(2)
+    end
+
+    it "will use the value at index[0] when not explicitly passed an initial value" do
+      expect([1].injector).to eq(1)
+    end
```

Comment:
You haven't checked against the real `#inject` - the two tests above would not pass with the real method